lib/providers/battery_level_provider.dart

```diff
@@ -9,13 +9,11 @@ class BatteryLevelProvider extends DeviceProvider {
       : super(request, deviceId, getDetails: getDetails);
 
   BatteryLevelTrait? getBatteryLevelTrait() {
-    return deviceDetail?.traits
-            .firstWhere((trait) => trait.runtimeType == BatteryLevelTrait)
-        as BatteryLevelTrait?;
+    return trait<BatteryLevelTrait>() as BatteryLevelTrait?;
   }
 
   int get getBatteryLevel {
-    return getBatteryLevelTrait()?.state.value ?? 0;
+    return state<BatteryLevelTrait, BatteryLevel>()?.value ?? 0;
```

Comment:
What do you think about using the convention `get[Name_of_State]State` for retrieving States? e.g. here it could be `getBatteryLevelState`

Reply:
I strongly agree with any naming convention but that one is a solid one lol.